### PR TITLE
extraction code for gens

### PR DESCRIPTION
**Pull request by Uzay-G**

### CODE REVIEW

File: front/pages/w/[wId]/u/gens/index.tsx [hunk 1]
```diff
@@ -177,10 +177,68 @@ export class FunctionSingleArgParser {
 
 export function DocumentView({
   document,
+  query,
+  owner,
 }: {
   document: GensRetrievedDocumentType;
+  query: string;
+  owner: any;
```
Comment (spolu) [inline front/pages/w/[wId]/u/gens/index.tsx:167]: let's type this guy should simply be `Owner`

File: front/pages/w/[wId]/u/gens/index.tsx [hunk 2]
```diff
@@ -177,10 +177,68 @@ export class FunctionSingleArgParser {
...
 }) {
   const provider = providerFromDocument(document);
+  const [extractedText, setExtractedText] = useState<string>("");
+
+  useEffect(() => {
+    let extractInput = [
+      {
+        query: query,
+        result: document,
```
Comment (spolu) [inline front/pages/w/[wId]/u/gens/index.tsx:176]: result is weird as an input name but that's a nit

File: front/pages/w/[wId]/u/gens/index.tsx [hunk 3]
```diff
@@ -254,6 +355,7 @@ export default function AppGens({
   const [retrievalLoading, setRetrievalLoading] = useState(false);
   const [retrieved, setRetrieved] = useState([]);
   const [dataSources, setDataSources] = useState(workspaceDataSources);
+  const [resultTexts, setResultTexts] = useState({}); // TODO: more efficient here
```
Comment (spolu) [inline front/pages/w/[wId]/u/gens/index.tsx]: unused right? let's remove

**Review by spolu — CHANGES_REQUESTED** [review]
Small fixes + there is a conflict to fix otherwise LGMT :+1:

File: front/pages/w/[wId]/u/gens/index.tsx (hunk 2 above)
Comment (Uzay-G) [reply front/pages/w/[wId]/u/gens/index.tsx:176]: It's looking at a searchresult, but I don't know I guess

**Review by spolu — APPROVED** [review]

**Pull request merged**